src/db/collections/outputs/mod.rs

```diff
@@ -175,9 +162,9 @@ impl MongoDb {
         self.db
             .collection::<OutputDocument>(OutputDocument::COLLECTION)
             .update_one(
-                doc! { "metadata.output_id": output_id },
-                doc! { "$set": bson::to_document(&OutputDocument::from(output))? },
-                UpdateOptions::builder().upsert(true).build(),
+                doc! { "_id": output_id },
```

Comment:
Not sure if it's okay to do this, because we will end up with dangling metadata with no underlying output data if we never inserted the unspent piece.